sdcm/utils/version_utils.py

```diff
@@ -4,65 +4,161 @@
 from pkg_resources import parse_version
 from repodataParser.RepoParser import Parser
 
+from sdcm.utils.common import ParallelObject
 
 # Examples of ScyllaDB version strings:
 #   - 666.development-0.20200205.2816404f575
 #   - 3.3.rc1-0.20200209.0d0c1d43188
 #   - 2019.1.4-0.20191217.b59e92dbd
+
+# gemini version 1.0.1, commit ef7c6f422c78ef6b84a6f3bccf52ea9ec846bba0, date 2019-05-16T09:56:16Z
+GEMINI_VERSION_RE = re.compile(r'\s(?P<gemini_version>([\d]+\.[\d]+\.[\d]+)?),')
+
 SCYLLA_VERSION_RE = re.compile(r"\d+(\.\d+)?\.[\d\w]+([.~][\d\w]+)?")
+PRIMARY_XML_GZ_REGEX = re.compile(r'="(.*?primary.xml.gz)"')
+
+REPOMD_XML_PATH = "repodata/repomd.xml"
+URL_REGEX_AND_FORMAT_LIST = [
+    (re.compile(r"baseurl=(?P<url>http.*scylladb.com.*)"), "{url}" + REPOMD_XML_PATH),
+    (re.compile(r"deb\s+(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$"),
+     "{url}/dists/{version_code_name}/{component}/binary-amd64/Packages.gz"),
+    (re.compile(r"deb\s+\[arch=(?P<arch>.*?)\]\s(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$"),
+     "{url}/dists/{version_code_name}/{component}/binary-{arch}/Packages"),
+]
+
+SCYLLA_URL_REGEX = re.compile(".*(?P<url>http.*scylladb.com.*).*")
+
+SCYLLA_URL_RESPONSE_TIMEOUT = 30
+SUPPORTED_XML_EXTENSIONS = ("xml", "xml.gz")
+SUPPORTED_FILE_EXTENSIONS = ("list", "repo", "Packages", "gz") + SUPPORTED_XML_EXTENSIONS
+BASE_ARCH_KEY = "$basearch"
+RELEASE_VERSION_KEY = "$releasever"
+URL_ENV_VARIABLES = {
+    BASE_ARCH_KEY: ["x86_64", "noarch"],
+    RELEASE_VERSION_KEY: ["7"],
+}
+VERSION_NOT_FOUND_ERROR = "URL isn't a correct deb list or yum repo"
+
+
+def get_all_scylla_url_patterns(url):
```

Comment:
this function is getting one url, why it is called `get_all_*` ?

Reply:
Although the function receives one URL, it extracts all the patterns that have `baseurl` prefix.
A file can contain multiple patterns (some of them not Scylla URL), so I called the function like that.